### PR TITLE
new example 000

### PR DESCRIPTION
**Pull request by agvarghe**

### CODE REVIEW

File: src/ansys/meshing/prime/examples/examples.py [hunk 1]
```diff
@@ -1010,7 +1016,7 @@ def download_f1_rw_drs_stl(
     >>>     model = session.model
     >>>     f1_rw_drs = prime_examples.download_f1_rw_drs_stl()
     >>>     with prime.FileIO(model) as io:
-    >>>         _ = io.read_pmdat(pcb, params=prime.FileReadParams(model))
+    >>>         _ = io.read_cad(f1_rw_drs, params=prime.FileReadParams(model))
```
Comment (waltersma) [inline src/ansys/meshing/prime/examples/examples.py:1019]: Hi @agvarghe, I think you need to modify this line as this should be import_cad and has associated params.  Same again for similar instances below.

File: src/ansys/meshing/prime/examples/examples.py [hunk 2]
```diff
@@ -1115,8 +1121,113 @@ def download_f1_rw_main_plane_stl(
...
+
+    Examples
+    --------
+    >>> import ansys.meshing.prime as prime
+    >>> import ansys.meshing.prime.examples as prime_examples
+    >>> with prime.launch_prime() as session:
+    >>>     model = session.model
+    >>>     block_model = prime_examples.download_block_model_scdoc()
+    >>>     with prime.FileIO(model) as io:
+    >>>         _ = io.import_cad(block_model, params=prime.FileReadParams(model))
```
Comment (waltersma) [inline src/ansys/meshing/prime/examples/examples.py:1159]: Hi @agvarghe, params not defined correctly.

**Review by waltersma — CHANGES_REQUESTED** [review]
examples need fixing.

File: examples/gallery/00_lucid_file_IO.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+mesh_util = prime.lucid.Mesh(model=model)
+
+###############################################################################
+# Import geometry
+# ~~~~~~~~~~~~~~~
+# Download the block CAD file named `pyprime_block_import.fmd` and
+# Import the geometry into pyprimemesh.
+# Display part details
+
+mesh_util.read(file_name=prime.examples.download_block_model_scdoc())
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:59]: @agvarghe this needs to be fmd or pmdat for running in a container as the SpaceClaim out of process option will not be available in this environment.

File: examples/gallery/00_lucid_file_IO.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+Exporting mesh in different file formats
+================================================================
+
+**Summary**: This example demonstrates the procedure for creating a mesh for a block model
+and saving the resulting mesh in various file formats.
+
+Objective
+~~~~~~~~~~
+
+The objective is to illustrate the process of exporting mesh files into a variety of
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:14]: @agvarghe objective is to illustrate how data is converted and passed during import and export of mesh and geometry?

File: examples/gallery/00_lucid_file_IO.py [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+.. image:: ../../../images/generic_solid_block.png
+   :align: center
+   :width: 800
+   :alt: Generic Solid Block.
+
+Procedure
+~~~~~~~~~~
+* Launch an Ansys Prime Server instance and instantiate the meshing utilities
+  from the ``lucid`` class.
+* Import the CAD geometry file for the solid block.
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:26]: @agvarghe Import CAD geometry and review the imported entities

File: examples/gallery/00_lucid_file_IO.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+* Generate surface mesh with a constant mesh size of 2mm.
+* Generate volume mesh using tetrahedral elements and default settings .
+* Export the mesh file as pmdat, cdb and cas format.
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:29]: @agvarghe + Review the entities to be exported to solvers.

File: examples/gallery/00_lucid_file_IO.py [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+* Import the mesh file.
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:30]: @agvarghe Import the created solver files to review the entities as they are coming from the solvers.

File: examples/gallery/00_lucid_file_IO.py [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+
+###############################################################################
+# Import geometry
+# ~~~~~~~~~~~~~~~
+# Download the CAD file named `pyprime_block_import.fmd` and
+# proceed to import its geometry into PyPrimeMesh.
+# Display part details by printing the model. **pyprime_block_import**
+# is the name of the part created. Topology exists in the form of the
+# following entities: Topo Edges, Topo Faces and Topo Volumes.
+# The topology data is preserved in the part when created from a cad file.
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:60]: @agvarghe explain what is there in SCDM and how this corresponds to the entities in Prime

File: examples/gallery/00_lucid_file_IO.py [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+#
+# Named Selections present in the CAD are imported as labels. In this case, the Named Selection
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:62]: @agvarghe should we define this as "Named selections or groups"

File: examples/gallery/00_lucid_file_IO.py [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+# present in the CAD **my_group** is imported as a label.
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:63]: @agvarghe lets add both an edge group and a volume group and show these in this example

File: examples/gallery/00_lucid_file_IO.py [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+# The bodies present in the CAD are imported as Volume Zones,  in this case the body **solid** is
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:64]: @agvarghe lets define Zones, "a collection of either topo or zonelet entities that we can assign properties to in a solver when exported as mesh".  and an example such as "if we wish to assign a material to a region of the model we can define a volume zone for multiple topo volumes or cell zonelets so we can apply the property".

File: examples/gallery/00_lucid_file_IO.py [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+# imported as a volume zone.
+
+mesh_util.read(file_name=prime.examples.download_block_model_fmd())
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:67]: @agvarghe add a commented line here to show reading the scdoc if windows OS and Ansys install is available

File: examples/gallery/00_lucid_file_IO.py (hunk 8 above)
Comment (waltersma) [reply examples/gallery/00_lucid_file_IO.py:60]: in PyPrimeMesh we have a single topo part with 1 topo volume, x topo faces and y topo edges.  These entities correspond to SCDM....
A Volume zone is also defined for the topovolume for the solid body.

File: examples/gallery/00_lucid_file_IO.py [hunk 13]
```diff
@@ -0,0 +1,252 @@
...
+print(model)
+
+###############################################################################
+# Generate Mesh
+# ~~~~~~~~~~~~~~~
+# The topo faces consists of geom zonelets and mesh zonelets,
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:73]: @agvarghe the topo part currently has no mesh associated with it and contains only geometry.

File: examples/gallery/00_lucid_file_IO.py [hunk 14]
```diff
@@ -0,0 +1,252 @@
...
+# after CAD import geom zonelets are created,
+# theu consist of the tesselations present in the CAD.
+# Using Lucid api, User can generate a conformal mesh , this is called meshed zonelets.
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:76]: meshed zonelets are only available once the topo part is converted to a mesh part by deleting the topo entities.

File: doc/source/images/00_after_cad_import.png
Comment (waltersma) [inline doc/source/images/00_after_cad_import.png:1]: @agvarghe please review the numbers of entities in these images as they look incorrect.  Topoedges seem to be confused with topofaces.

**Review by waltersma — CHANGES_REQUESTED** [review]

File: doc/source/images/00_after_cad_import.png
Comment (agvarghe) [reply doc/source/images/00_after_cad_import.png:1]: Thanks, i have fixed this in the latest commit

File: examples/gallery/00_lucid_file_IO.py [hunk 15]
```diff
@@ -0,0 +1,275 @@
...
+# The topology consists of the following TopoEntities , they are TopoEdges, TopoFaces
+# and TopoVolumes.
+#
+# * TopoEdge represent the curves/edges present in the CAD.
+#   In this case there are **08 edges** present in SpaceClaim are imported
+#   as **08 TopoEdges**.
+#
+# * TopoFace represent the surfaces/faces present in the CAD.
+#   The **17 CAD Faces** present in SpaceClaim are imported as **17 Topofaces** in
+#   PyPrimeMesh.
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:81]:
```suggestion
# * TopoEdge represent the curves/edges present in the CAD.
#   In this case there are **17 edges** present in SpaceClaim are imported
#   as **17 TopoEdges**.
#
# * TopoFace represent the surfaces/faces present in the CAD.
#   The **8 CAD Faces** present in SpaceClaim are imported as **8 Topofaces** in
#   PyPrimeMesh.
```

**Comment by waltersma** [comment]
@ninad-kamat @agvarghe for the cdb import we may want to delay this part of the example until this issue is resolved.

**Comment by agvarghe** [comment]
> @ninad-kamat @agvarghe for the cdb import we may want to delay this part of the example until this issue is resolved.

sure, i will modify the file and remove the section talking about cdb import

File: examples/gallery/00_lucid_file_IO.py [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+
+mesh_util.surface_mesh(min_size=2.0)
+display = Graphics(model)
+display()
+part = model.get_part_by_name("pyprime_block_import")
+display(scope=prime.ScopeDefinition(model, label_expression="my_group"))
+mesh_util.volume_mesh()
+
+###############################################################################
+# Export mesh  as PyPrimeMesh (.pmdat) native format mesh file
```
Comment (waltersma) [inline examples/gallery/00_lucid_file_IO.py:129]:
```suggestion
# Export mesh as PyPrimeMesh (.pmdat) native format mesh file
```

**Pull request merged**